### PR TITLE
Dulaya WIP: Navigation To FoodDetail and to Cart

### PR DESCRIPTION
**Pull request by aahenry51**
To Do:
 - Create add-ons (w prices) to data in server ✔️
 - Render add-ons ✔️
 - Render rating and rating count ✔️
 - Create another component for add-ons ✔️
 - Remove logic from UI ✔️
 - Recalculate price based on if add-on is selected or not ✔️
 - Send all data and calculated price to Cart Screen ✔️
   - Possibly need Global State because data shouldn't pass via navigation, only params ❌
 - Navigate to Cart Screen when Add To Cart is pressed ✔️
 - Navigate to Review when See Review is clicked ✔️
 - Optional: toggle right icon (Heart) ❌
 - Fix type errors ✔️
image

### CODE REVIEW

**Comment by aahenry51** [comment]
It goes from:
Home -> FoodDetails
Category -> FoodDetails
Favorites -> FoodDetails
FoodDetails -> Cart

**Review by mshago — APPROVED** [review]

**Comment by aahenry51** [comment]
I want to edit something with the category screen

**Comment by carizzuti** [comment]
The Food Details screen should also navigate to the reviews screen when clicking 'See Review'

File: mobile/src/screens/foodDetails/FoodDetails.type.ts [hunk 1]
```diff
@@ -22,6 +22,7 @@ export type TFoodDetailsScreenViewProps = {
   countPlusMinusChange: () => {};
   onPressPlus: () => {};
   onPressMinus: () => {};
+  onPressAddToCart: () => null;
   addOn: [];
```
Comment (lasharela) [inline mobile/src/screens/foodDetails/FoodDetails.type.ts:26]: not just []

File: mobile/src/screens/home/Home.tsx [hunk 2]
```diff
@@ -1,27 +1,29 @@
 import React, {FC, useEffect, useState} from 'react';
 import {Alert} from 'react-native';
-import { foodHubAPI } from '../../config';
+import {foodHubAPI} from '../../config';
 import {TGetItemId, THomeData, THomeScreenNavigation} from './Home.type';
+import {EAppNavigationRoutes} from '../../navigation/appNavigation/AppNavigation.type';
 import {HomeScreenView} from './Home.view';
 
 export const HomeScreen: FC<THomeScreenNavigation> = ({navigation}) => {
   const [category, setCategory] = useState<null | string>(null);
```
Comment (lasharela) [inline mobile/src/screens/home/Home.tsx:9]: const [category, setCategory] = useState<string>();
typeof category = string | undefined

**Review by lasharela — APPROVED** [review]

**Pull request merged**